File: backend/internal/models/notice.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Notice struct {
	ID        uuid.UUID  `json:"id"`
	Title     string     `json:"title"`
	Content   string     `json:"content"`
	PostedBy  *uuid.UUID `json:"posted_by,omitempty"`
	IsPinned  bool       `json:"is_pinned"`
	IsActive  bool       `json:"is_active"`
	Priority  int        `json:"priority"`
	CreatedAt time.Time  `json:"created_at"`
	UpdatedAt time.Time  `json:"updated_at"`
}

type CreateNoticeRequest struct {
	Title    string `json:"title"`
	Content  string `json:"content"`
	IsPinned bool   `json:"is_pinned,omitempty"`
}

type UpdateNoticeRequest struct {
	Title    string `json:"title,omitempty"`
	Content  string `json:"content,omitempty"`
	IsPinned *bool  `json:"is_pinned,omitempty"`
	IsActive *bool  `json:"is_active,omitempty"`
	Priority *int   `json:"priority,omitempty"`
}

File: frontend/src/types/notice.ts
export interface Notice {
  id: string;
  title: string;
  content: string;
  posted_by?: string;
  is_pinned: boolean;
  is_active: boolean;
  priority: number;
  created_at: string;
  updated_at: string;
}

export interface CreateNoticeRequest {
  title: string;
  content: string;
  is_pinned?: boolean;
}

export interface UpdateNoticeRequest {
  title?: string;
  content?: string;
  is_pinned?: boolean;
  is_active?: boolean;
  priority?: number;
}